spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialMaxPooling.scala

```diff
@@ -87,6 +87,39 @@ class SpatialMaxPooling[T: ClassTag](
     this
   }
 
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toSingle().toArray
+    require(input.length == 4,
+      "SpatialMaxPooling: " + ErrorInfo.constrainInputAs3DOrBatch)
+    val (dimh, dimw, dimc) = format.getHWCDims(input.length)
+    val nInputPlane = input(dimc -1)
+    val inputHeight = input(dimh -1)
+    val inputWidth = input(dimw -1)
+    val sizes =
+      if (padW == -1 && padH == -1) {
+        Utils.getSAMEOutSizeAndPadding(inputHeight, inputWidth, dH, dW, kH, kW)
+      } else {
+        require(inputWidth >= kW - padW && inputHeight >= kH - padH,
+          "input smaller than kernel size. " +
+            s"current input size($inputWidth, $inputHeight), " +
+            s"kernel size(${kW-padW}, ${kH-padH})")
+        require(kW / 2 >= padW && kH / 2 >= padH,
+          "pad should be smaller than half of kernel size. " +
+          s"current pad size($padW, $padH), " + s"kernel size($kW, $kH)")
+        Utils.getOutSizeAndPadding(inputHeight, inputWidth, dH, dW, kH, kW, padH, padW, ceilMode)
+      }
+    val oHeight = sizes(4)
+    val oWidth = sizes(5)
+
+    val outputShape = format match {
+      case DataFormat.NCHW =>
```

Comment:
remember to add unitest for both format

Reply:
Already there. https://github.com/intel-analytics/BigDL/pull/2214/files#diff-710faeb0d79adba6f8f90c251eff80bc